args.zig

```diff
@@ -256,25 +264,46 @@ fn parseInternal(comptime Generic: type, comptime MaybeVerb: ?type, args_iterato
                     inline for (std.meta.fields(Verb)) |fld| {
                         if (std.mem.eql(u8, item, fld.name)) {
                             // found active verb, default-initialize it
-                            result.verb = @unionInit(Verb, fld.name, fld.field_type{});
+                            result.verb = @unionInit(Verb, fld.name, undefined);
+                            const target_value = &@field(result.verb.?, fld.name);
+
+                            const VerbType = fld.field_type;
+                            inline for (std.meta.fields(VerbType)) |field| {
+                                if (field.default_value) |default_value_ptr| {
+                                    const default_value = @ptrCast(*const field.field_type, default_value_ptr).*;
+                                    @field(target_value, field.name) = default_value;
+                                } else {
+                                    try required_map.put(field.name, true);
+                                }
+                            }
+
+                            continue :args_loop;
                         }
                     }
 
-                    if (result.verb == null) {
-                        try error_handling.process(error.EncounteredUnknownVerb, Error{
-                            .option = "verb",
-                            .kind = .unsupported,
-                        });
-                    }
+                    try error_handling.process(error.EncounteredUnknownVerb, Error{
+                        .option = "verb",
+                        .kind = .unsupported,
+                    });
 
                     continue;
                 }
             }
 
+            // Argument doesn't match anything, so should be a argument to the program itself
             try arglist.append(try result_arena_allocator.dupeZ(u8, item));
         }
     }
 
+    if (required_map.count() > 0) {
+        last_error = error.MissingRequiredArgument;
+        try error_handling.process(error.MissingRequiredArgument, Error{
```

Comment:
Good point. Right now I'm printing it to a buffer and setting the 'option' of ErrorHandling so we use that mechanism. the printing is not so pretty, right now I just print the arguments between '[' ']' since there can be multiple.

Reply:
I guess calling `error_handling.process` several times is a much more elegant solution here (as this is already done above) and each prints its own line

Reply:
Yeah that works too, I changed it.